### PR TITLE
Allow toast dismissal without container

### PR DESCRIPTION
**Pull request by trylle**
This is a suggested minimal fix+test case for #811.

### CODE REVIEW

File: src/core/toast.ts [hunk 1]
```diff
@@ -195,7 +195,13 @@ toast.dark = (content: ToastContent, options?: ToastOptions) =>
 /**
  * Remove toast programmaticaly
  */
-toast.dismiss = (id?: Id) => eventManager.emit(Event.Clear, id);
+toast.dismiss = (id?: Id) => {
+  if (containers.size > 0) {
+    eventManager.emit(Event.Clear, id);
+  } else {
+    queue = queue.filter(t => id !== undefined && t.options.toastId !== id);
```
Comment (fkhadra) [inline src/core/toast.ts:202]: I would replace `id !== undefined` by `id != undefined`. Just to check for both undefined and null.
```suggestion
    queue = queue.filter(t => id != undefined && t.options.toastId !== id);
```

Comment (fkhadra) [reply src/core/toast.ts:202]: Actually, even this code will do the work 🤔? WDYT?
```suggestion
    queue = queue.filter(t => t.options.toastId !== id);
```

Comment (trylle) [reply src/core/toast.ts]: I was trying to preserve the behavior of `toast.dismiss`. I gathered that this function would remove all toasts if you did not provide an id. Therefore the whole queue will be emptied if `id` is undefined.

Comment (trylle) [reply src/core/toast.ts]: It might be better/more consistent to use `isToastIdValid`, though?

**Review by fkhadra — APPROVED** [review]

File: src/core/toast.ts (hunk 1 above)
Comment (fkhadra) [reply src/core/toast.ts]: Indeed you're right.

**Comment by slashwhatever** [comment]
While this has been merged and released, I just wanted to flag that a similar sounding behaviour is in the latest release 9.1.3

I've been struggling with a similar issue (only in Firefox) where a toast when dismissed programmatically will remain on screen despite actually calling the dismiss method with a valid id. I assume this is somehow potentially related to the container issue.

Reverting to 9.0.8 has fixed this for me. 

One other thing noted in my testing; the toastId format has changed from a complex alpha string to a simple number string. not sure if related. Just thought it was odd.

**Pull request merged**